### PR TITLE
source code improvements + attributes adjustement

### PR DESCRIPTION
**Pull request by konraddysput**
## Why

This diff allows us to get a relative application path instead of an absolute application path in stack trace property. This diff also contains improvements for node processes like Electron applications that have limitations with accessing for example package.json or modules package.json.

## Test cases
Integration was tested against the latest version of the electron application - in a sandboxed environment. Sample JSON fragment that we're sending to Backtrace that proofs that we're now providing relative path instead of an absolute path.

```
  "threads": {
        "main": {
            "name": "main",
            "fault": true,
            "stack": [
                {
                    "funcName": "ElectronExample.ipcReply",
                    "sourceCode": "C:\\Users\\konra\\source\\repos\\backtrace-node\\examples\\typescript\\electron\\backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "path": "backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "line": 160,
                    "column": 32
                },
                {
                    "funcName": "ElectronExample.forwardMessage",
                    "sourceCode": "C:\\Users\\konra\\source\\repos\\backtrace-node\\examples\\typescript\\electron\\backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "path": "backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "line": 157,
                    "column": 14
                },
                {
                    "funcName": "ElectronExample.<anonymous>",
                    "sourceCode": "C:\\Users\\konra\\source\\repos\\backtrace-node\\examples\\typescript\\electron\\backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "path": "backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "line": 143,
                    "column": 30
                },
                {
                    "funcName": "step",
                    "sourceCode": "C:\\Users\\konra\\source\\repos\\backtrace-node\\examples\\typescript\\electron\\backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "path": "backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "line": 52,
                    "column": 23
                },
                {
                    "funcName": "Object.next",
                    "sourceCode": "C:\\Users\\konra\\source\\repos\\backtrace-node\\examples\\typescript\\electron\\backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "path": "backtrace-ts-electron-sample-win32-x64\\resources\\app\\dist\\main.js",
                    "line": 33,
                    "column": 53
                },
                ```

### CODE REVIEW

**Review by jasoncdavis0 — APPROVED** [review]

File: source/model/backtraceStackTrace.ts [hunk 1]
```diff
@@ -208,9 +211,7 @@ export class BacktraceStackTrace {
     });
   }
 
-  private isCallingModule(stackFrame: IBacktraceStackFrame): boolean {
-    return (
-      !this.callingModulePath && fs.existsSync(stackFrame.sourceCode) && !stackFrame.sourceCode.includes('node_modules')
-    );
+  private isCallingModule(sourcePath: string): boolean {
+    return !!sourcePath && !this.callingModulePath && !sourcePath.includes('node_modules');
```
Comment (rqbacktrace) [inline source/model/backtraceStackTrace.ts:215]: Did you mean to put this `!!` here? I'm not sure how to interpret this logic

Comment (konraddysput) [reply source/model/backtraceStackTrace.ts:215]: Yes - it's expected. 
Let's talk about two use cases:
#1 `sourcePath` is `undefined`/`null`/`empty
#2 `sourcePath` is defined

for #1 
!sourcePath => true 
!!sourcePath => false
 
for #2 
!sourcePath => false
!!sourcePath => true

we want to check by using `!!` if object is defined.

Comment (rqbacktrace) [reply source/model/backtraceStackTrace.ts:215]: Okay sounds good, so `isCallingModule` returns true if:
sourcePath **is** defined, callingModulePath **is not** defined, and sourcePath **is not** a node_module

Comment (rqbacktrace) [reply source/model/backtraceStackTrace.ts:215]: That makes sense and it matches the previous logic too

**Review by rqbacktrace — APPROVED** [review]
Looks good to me!

**Pull request merged**